docs/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md

```diff
@@ -226,29 +226,81 @@ Users can generate access to the Amazon EKS cluster via the `AWS CLI`.
+        }
+      }
+    }
+  }
+}
+```
+
+In this updated configuration:
+
+- `principal_arn` should be replaced with the ARN of the IAM user or role.
+- `policy_associations` allow you to associate policies for fine-grained access control.
+
+For a list of policies, please visit the [AWS EKS Access Policies documentation](https://docs.aws.amazon.com/eks/latest/userguide/access-policies.html).
+
+:::info
+
+Please note that the version 2.x.x of this module no longer supports direct mappings via `aws_auth_roles` and `aws_auth_users`. If you are upgrading from version [1.x.x](https://github.com/camunda/camunda-tf-eks-module/releases/tag/1.0.3), fork the module repository and follow the official AWS instructions for managing the `aws-auth` ConfigMap.
+For more details, refer to the [official upgrade guide](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/UPGRADE-20.0.md).
+
+:::
+
+### Terraform AWS IAM Permissions
+
+The user creating the Amazon EKS cluster has admin access by default.
+To manage user access use the `access_entries` configuration introduced in module version [2.0.0](https://github.com/camunda/camunda-tf-eks-module/releases/tag/2.0.0):
 
 ```hcl
-# AWS IAM roles mapping
-aws_auth_roles = [{
-    rolearn  = "<arn>"
-    username = "<username>"
-    groups   = ["system:masters"]
-  }]
-
-# AWS IAM users mapping
-aws_auth_users = [{
-    userarn  = "<arn>"
-    username = "<username>"
-    groups   = ["system:masters"]
-  }]
+access_entries = {
+  example = {
+    kubernetes_groups = []
+    principal_arn     = "<arn>"
+
+    policy_associations = {
+      example = {
+        policy_arn   = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSViewPolicy"
+        access_scope = {
+          namespaces = ["default"]
+          type       = "namespace"
+        }
+      }
+    }
+  }
+}
 ```
 
-Where `arn` is the `arn` of your user or the role. The `group` is the Kubernetes rule, where `system:masters` is equivalent to an admin role. Lastly, `username` is either the username itself or the role name, which is used for logs.
+In this updated configuration:
+
+- `principal_arn` should be replaced with the ARN of the IAM user or role.
+- `policy_associations` allow you to associate policies for fine-grained access control.
+
+For a list of policies, please visit the [AWS EKS Access Policies documentation](https://docs.aws.amazon.com/eks/latest/userguide/access-policies.html).
+
+:::info
+
+Please note that the version 2.x.x of this module no longer supports direct mappings via `aws_auth_roles` and `aws_auth_users`. If you are upgrading from version [1.x.x](https://github.com/camunda/camunda-tf-eks-module/releases/tag/1.0.3), fork the module repository and follow the official AWS instructions for managing the `aws-auth` ConfigMap.
+For more details, refer to the [official upgrade guide](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/UPGRADE-20.0.md).
+
+:::
```

Comment:
I think a copy and paste made it's way into this as it's the exact same thing as the previous lines 229-265